### PR TITLE
[WIP] Sbromberger/connectivity v1.4.0

### LINKED ISSUE #1363: bug in neighborhood function
```
g = SimpleGraph(3)
add_edge!(g, 1, 2)
add_edge!(g, 1, 3)
add_edge!(g, 3, 2)
dismx = zeros((3, 3))
dismx[1, 2] = 10
dismx[1, 3] = 1
dismx[3, 2] = 1
neighborhood(g, 1, 4, dismx)

```
will result in `[1, 3]` despite 2 can be reached from 1 by a path of length 2 .

### PR DESCRIPTION
**Pull request by sbromberger**
This code migrates `connectivity.jl` into its own module, and fixes #1363. We'll need to cherry-pick or otherwise migrate #1362 as well.

### CODE REVIEW

**Comment by sbromberger** [comment]
@simonschoelly @matbesancon review please? Thanks.

**Comment by sbromberger** [comment]
codecov is in the legacy (deprecated) code, and I can't figure out why the deprecations aren't being tested. Minor decrease anyway.

**Comment by sbromberger** [comment]
I plan on merging this on March 27.

**Comment by matbesancon** [comment]
haven't forgotten this one, reading it this evening

File: src/Connectivity/Connectivity.jl [hunk 1]
```diff
@@ -0,0 +1,646 @@
+module Connectivity
```
Comment (matbesancon) [inline src/Connectivity/Connectivity.jl]: I would add a docstring to the module itself to document what it regroups

File: src/Connectivity/Connectivity.jl [hunk 2]
```diff
@@ -0,0 +1,646 @@
...
+    return d
+end
+
+"""
+    components(labels)
+
+Given a vector of component labels, return a vector of vectors representing the vertices associated
+with a given component id.
+"""
+function components(labels::Vector{T}) where T <: Integer
```
Comment (matbesancon) [inline src/Connectivity/Connectivity.jl]:
```suggestion
function components(labels::AbstractVector{T}) where {T <: Integer}
```
Vector not required, also changed the formatting for consistency but totally optional

File: src/Connectivity/Connectivity.jl [hunk 3]
```diff
@@ -0,0 +1,646 @@
...
+julia> g = SimpleDiGraph(Edge.(edge_list))
+{11, 13} directed simple Int64 graph
+
+julia> strongly_connected_components(g)
+4-element Array{Array{Int64,1},1}:
+ [8, 9]
+ [5, 6, 7]
+ [1, 2, 3, 4]
+ [10, 11]
+
```
Comment (matbesancon) [inline src/Connectivity/Connectivity.jl]:
```suggestion
```
Last line of the block not required

File: src/Connectivity/Connectivity.jl [hunk 4]
```diff
@@ -0,0 +1,646 @@
...
+```
+"""
+
```
Comment (matbesancon) [inline src/Connectivity/Connectivity.jl]:
```suggestion
```

File: src/Connectivity/Connectivity.jl [hunk 5]
```diff
@@ -0,0 +1,646 @@
...
+function strongly_connected_components end
+# see https://github.com/mauro3/SimpleTraits.jl/issues/47#issuecomment-327880153 for syntax
+@traitfn function strongly_connected_components(g::AG::IsDirected) where {T<:Integer, AG <: AbstractGraph{T}}
+    zero_t = zero(T)
+    one_t = one(T)
+    nvg = nv(g)
+    count = one_t
+
```
Comment (matbesancon) [inline src/Connectivity/Connectivity.jl]:
```suggestion
```
Two empty lines not necessary

File: test/Connectivity/runtests.jl [hunk 6]
```diff
@@ -0,0 +1,18 @@
+using LightGraphs.ShortestPaths
+using LightGraphs.Traversals
+
+const LC = LightGraphs.Connectivity
+
+const conntestdir = dirname(@__FILE__)
+
+conntests = [
```
Comment (matbesancon) [inline test/Connectivity/runtests.jl]:
```suggestion
const conntests = [
```

File: src/Connectivity/Connectivity.jl (hunk 1 above)
Comment (sbromberger) [reply src/Connectivity/Connectivity.jl]: @kshyatt and I will be doing a doc overhaul prior to 2.0.

Comment (matbesancon) [reply src/Connectivity/Connectivity.jl]: Still I guess we can add something

Comment (sbromberger) [reply src/Connectivity/Connectivity.jl]: I'd really rather do it consistently once we've made decisions as to where things are going to go. I anticipate some movement of functions among different modules.

Comment (sbromberger) [reply src/Connectivity/Connectivity.jl]: I'll open an issue to make sure we get to this, though. Thanks for bringing it up.

Comment (sbromberger) [reply src/Connectivity/Connectivity.jl]: Ref #1376

**Comment by sbromberger** [comment]
I don't know why travis isn't working.

**Comment by sbromberger** [comment]
one commit to kick travis.

**Comment by sbromberger** [comment]
This is the old travis that's not working, so I think we're good here. Merging.

**Pull request merged**